src/remote.c

```diff
@@ -757,10 +771,16 @@ int git_remote__connect(git_remote *remote, git_direction direction, const git_r
 
 	remote->transport = t;
 
+	free(url);
+
 	return 0;
 
 on_error:
-	t->free(t);
+	if (t)
+		t->free(t);
+
+	if (url)
+		free(url);
```

Comment:
No need for the `if`

Reply:
Mhm. I wonder where my misconception comes from that `free(NULL)` would be illegal... I'm obviously traumatized by some other language.